### PR TITLE
Update Doc and include Release doc information

### PR DESCRIPTION
**Pull request by aleskandro**
- Introduce ocp-release.doc
- Update readme, godoc and CRDs based on the recent documentation changes

Related to https://issues.redhat.com/browse/MULTIARCH-4290

Related to openshift/openshift-docs#77701

### CODE REVIEW

File: docs/ocp-release.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+The current approach is that for every new OCP release, we will have to:
+1. Create New Konflux Applicaiton and Component for the FBC
```
Comment (Prashanth684) [inline docs/ocp-release.md]: nit: `Applicaiton` to `Application`

Comment (aleskandro) [reply docs/ocp-release.md]: Don't consider for now, they are just notes on the fly yet.

File: docs/ocp-release.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+### Pre-release requirements
+
+- If the release is a major version or introduced significant
+[architectural changes](https://docs.engineering.redhat.com/pages/viewpage.action?pageId=402429315), 
+ensure that the threat model is reviewed and updated accordingly.
+
+### Release the operator
+
+1. Choose a snapshot from the list of snapshots and double check post-submits passed for it.
```
Comment (aleskandro) [inline docs/ocp-release.md:3]: @arewm can you have a look at this to double check if we are missing anything in particular to execute the release?

Comment (aleskandro) [reply docs/ocp-release.md:3]: cc @ralphbean too

File: docs/ocp-release.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    name: multiarch-tuning-operator-bundle
+    source:
+      git:
+        context: ./
+        dockerfileUrl: bundle.Dockerfile
+        revision: d73959c925629f29f9071fd6e7d58a0f58a54399
+        url: https://github.com/openshift/multiarch-tuning-operator
+[...]
+```
+3. Ensure that the containerImage of the operator is the one referenced by the bundle in the selected snapshot
```
Comment (ralphbean) [inline docs/ocp-release.md:30]: Good to have this step manual for now.

Watch https://issues.redhat.com/browse/KONFLUX-72. When it is done then you may no longer need to verify this manually.

File: docs/ocp-release.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+```shell
+podman create quay.io/redhat-user-workloads/multiarch-tuning-ope-tenant/multiarch-tuning-operator/multiarch-tuning-operator-bundle@sha256:...
+yq '.spec.install.spec.deployments[0].spec.template.metadata.annotations."multiarch.openshift.io/image"' /tmp/csv.yaml 
+```
+5. Create a new release for the operator in the Konflux cluster:
```
Comment (ralphbean) [inline docs/ocp-release.md]: Add a step to watch the `status` of the Release CR or look in the Konflux UI to confirm that images and bundle were published as expected?

File: docs/ocp-release.md (hunk 2 above)
Comment (ralphbean) [reply docs/ocp-release.md:3]: Looks good! I added a few minor suggestions. +1

Comment (aleskandro) [reply docs/ocp-release.md:3]: Thanks @ralphbean!

File: docs/ocp-release.md (hunk 4 above)
Comment (aleskandro) [reply docs/ocp-release.md]: Done

File: docs/ocp-release.md (hunk 3 above)
Comment (aleskandro) [reply docs/ocp-release.md:30]: Ack. Watching

**Comment by aleskandro** [comment]
/override "Red Hat Konflux / multiarch-tuning-operator-enterprise-contract / multiarch-tuning-operator" "Red Hat Konflux / multiarch-tuning-operator-enterprise-contract / multiarch-tuning-operator-bundle"

hope these are fixed by the next builds

**Pull request merged**